cmd/soroban-rpc/lib/preflight/src/lib.rs

```diff
@@ -664,3 +668,36 @@ fn extract_error_string<T>(simulation_result: &Result<T>, go_storage: &GoLedgerS
         }
     }
 }
+
+// xdr_to_json takes in a string name of an XDR type in the Stellar Protocol
+// (i.e. from the stellar_xdr crate) as well as a raw byte structure and returns a
+// JSONified string of said structure.
+//
+// On error, it will return a string with a JSON structure and an "error" key containing
+// the error message from Rust as well as a "type" key for the XDR type that it
+// attempted to decode from the given buffer (this should match `typename`).
+#[no_mangle]
+pub extern "C" fn xdr_to_json(typename: *mut libc::c_char, xdr: CXDR) -> *mut libc::c_char {
```

Comment:
question on organizational aspect, the xdr-json conversion isn't specific to preflight lib correct? is it worthwhile for conversion to be in a separate `lib/conversion`?

Reply:
It's not specific to preflight, but it does depend on some shared functionality like the `CXDR` structure and misc. memory management. In an ideal world we'd probably want to split that out into a shared lib but that seems a little preemptive for just a single function, since there'd be a bunch of busy work to make sure compiling/linking all works together across three separate Rust libs.